lisa/wa.py

```diff
@@ -645,3 +645,38 @@ def get_stats(self, **kwargs):
             agg_cols=['iteration', 'frame_id'],
             **kwargs,
         )
+
+class WASysfsExtractorCollector(WAArtifactCollectorBase):
+    NAME = 'sysfs-extractor'
+
+    def __init__(self, wa_output, path, type='diff', df_postprocess=None):
+        allowed_types = ['diff', 'before', 'after']
+
+        if type not in allowed_types:
+            self.logger.warn(
+                f'SysfsExtractor type must be one of {allowed_types}')
+            type = 'diff'
+
+        self._ARTIFACT_NAME = path + ' [' + type + ']'
+
+        # TODO: WA's sysfs-extractor augmentation can actually use a path
+        # to collect several files not only one...
+        self._filename = os.path.basename(path)
+
+        super().__init__(wa_output, df_postprocess=df_postprocess)
+
+    def _get_artifact_df(self, path):
+        # WA given path is actually dirname
+        path = os.path.join(path, self._filename)
+
+        with open(path, 'r') as f:
+            raw_file = f.readlines()
+
+        raw_file = [line.strip() for line in raw_file]
```

Comment:
If the format is simply one numeric value per line, `pd.read_csv` would be faster and probably lower the peak memory usage

Reply:
It's supposed to read any type of sysfs/procfs file so the format can be anything.

Reply:
Can you add in comment in the file a snippet of "raw_file" please ? It's useful to know what is to be expected, and if it's something like `path/to/file: the content` we probably want to parse it and separate the path and content part in 2 columns.